src/main/java/org/broadinstitute/hellbender/tools/copynumber/CondenseReadCounts.java

```diff
@@ -0,0 +1,203 @@
+                    currentReadCount += record.getCount();
+                    lastPosition = record.getEnd();
+                    if (++currentRecordCount == factor) {
+                        if (expectedBinLength > 0 && (lastPosition - firstPosition + 1) == expectedBinLength) {
+                            if (condensedRecords == 0) {
+                                writer.writeHeaderComments(reader.headerLines());
+                            }
+                            writer.writeRecord(composeRecord(contig, firstPosition, lastPosition, currentReadCount));
+                            condensedRecords += currentRecordCount;
+                            outputCounts += currentReadCount;
+                        } else {
+                            skippedReadCounts += currentReadCount;
+                            skippedRecords += currentRecordCount;
+                        }
+                        contig = null;
+                        currentRecordCount = 0;
+                        currentReadCount = 0;
+                        firstPosition = lastPosition = -1;
+                    }
+                } else {
+                    skippedRecords += currentRecordCount;
+                    skippedReadCounts += currentReadCount;
+                    contig = record.getContig();
+                    firstPosition = record.getStart();
+                    lastPosition = record.getEnd();
+                    currentReadCount = record.getCount();
+                    currentRecordCount = 1;
+                }
+
+            }
+            skippedRecords += currentRecordCount;
+            skippedReadCounts += currentReadCount;
+            logger.info("Number of output records: " + condensedRecords / factor);
+            logger.info("Number of condensed input records: " + condensedRecords);
+            logger.info("Number of skipped input records: " + skippedRecords);
+            logger.info("Total count condensed: " + outputCounts);
+            logger.info("Total count skipped: " + skippedReadCounts);
+            logger.info(String.format("%% of lossed records and counts are: %.2g and %.2g",
+                    100.00 *(skippedRecords/ (double)(skippedRecords + condensedRecords)),
+                    100.00 *(skippedReadCounts/ (skippedReadCounts + outputCounts))));
+
+            return "SUCCESS";
+        } catch (final IOException ex) {
+            throw new GATKException(ex.getMessage(), ex);
+        }
+    }
+
+    private SimpleCount composeRecord(final String contig, final int start, final int end, final int count) {
+        return new SimpleCount(new SimpleInterval(contig, start, end), count);
+    }
+
+    private class SimpleCountReader extends TableReader<SimpleCount> {
+        private SimpleCountReader(final String input) throws IOException {
+            super(input, new InputStreamReader(BucketUtils.openFile(input)));
```

Comment:
Enabling read/write to buckets and bypassing creation of the list of objects in the CNV collection classes seem like two separate issues that should be solved by modifying the parent classes.  Probably only the second issue really affects cost/memory issues, since the files are not so big that localization/delocalization adds any overhead, correct?

I'm fine with doing this only for TSV count files to start, but other files could benefit.

Reply:
Actually, I'm not even convinced we need to work too hard to enable streaming in this tool, since it won't be long running or memory intensive.  I'd rather pay the localization/delocalization cost than the code cost.

Reply:
I agree, let's use `SimpleCountCollection` to handle the counts. We can go back and modify those for streaming performance later.

Reply:
Streaming does not make the code more complicate ... the more complex code (the conderser loop) would still apply with a fully load collection. 

About NIO ... BucketUtils make it kinda transparent whether it is actual NIO or a local file... basically we change "BuckedUtils.openFile" by  "new FileInputStream(". Also has the added benefit of handling ".gzs"

Reply:
You only need this custom SimpleCountReader so you can stream, correct?

It's great that BucketUtils makes it easy to enable streaming, so maybe we can enable this at the TableWriter level where it would have the most benefit.

Reply:
Also, just sticking to the vanilla SimpleCountCollection read/write allows the tool to operate on HDF5.